app/models/course/assessment.rb

```diff
@@ -13,6 +13,9 @@ class Course::Assessment < ActiveRecord::Base
   after_initialize :set_defaults, if: :new_record?
   before_validation :propagate_course
   before_validation :assign_folder_attributes
+  before_validation :prevent_reminder_duplication
+  before_save :setup_opening_reminders, if: :start_at_changed?, unless: :draft?
```

Comment:
Should be `after_save` ? so when there's a saving failure the notification does not get sent.